Seems like `-Wsuggest-override` catches a lot more cases than Clang's `-Winconsistent-missing-override`:
```
g++ -o platform/linuxbsd/godot_linuxbsd.linuxbsd.tools.64.o -c -std=gnu++17 -Wctor-dtor-privacy -Wnon-virtual-dtor -Wplacement-new=1 -Wno-error=cpp -g3 -pipe -fpie -Wall -Wextra -Wwrite-strings -Wno-unused-parameter -Wshadow-local -Wsuggest-override -Wno-misleading-indentation -Walloc-zero -Wduplicated-branches -Wduplicated-cond -Wstringop-overflow=4 -Wlogical-op -Wattribute-alias=2 -Werror -DDEBUG_ENABLED -DDEBUG_MEMORY_ALLOC -DDISABLE_FORCED_INLINE -DTOUCH_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -D_REENTRANT -DJOYDEV_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DVULKAN_ENABLED -DPTRCALL_ENABLED -DTOOLS_ENABLED -DMINIZIP_ENABLED -DZSTD_STATIC_LINKING_ONLY -DHAVE_MNTENT -Ithirdparty/freetype/include -Ithirdparty/libpng -Ithirdparty/vulkan -Ithirdparty/vulkan/include -Ithirdparty/vulkan/loader -Ithirdparty/zstd -Ithirdparty/zlib -Iplatform/linuxbsd -I. platform/linuxbsd/godot_linuxbsd.cpp
In file included from ./core/method_bind.h:391,
                 from ./core/class_db.h:34,
                 from ./core/resource.h:34,
                 from ./core/input/input_event.h:36,
                 from ./core/input/input.h:34,
                 from platform/linuxbsd/os_linuxbsd.h:34,
                 from platform/linuxbsd/godot_linuxbsd.cpp:37:
./core/method_bind.gen.inc:12:24: error: 'virtual Variant::Type MethodBind0::_gen_argument_type(int) const' can be marked override [-Werror=suggest-override]
   12 |  virtual Variant::Type _gen_argument_type(int p_arg) const { return _get_argument_type(p_arg); }
      |                        ^~~~~~~~~~~~~~~~~~
./core/method_bind.gen.inc:13:34: error: 'virtual GodotTypeInfo::Metadata MethodBind0::get_argument_meta(int) const' can be marked override [-Werror=suggest-override]
   13 |  virtual GodotTypeInfo::Metadata get_argument_meta(int p_arg) const {
      |                                  ^~~~~~~~~~~~~~~~~
./core/method_bind.gen.inc:25:23: error: 'virtual PropertyInfo MethodBind0::_gen_argument_type_info(int) const' can be marked override [-Werror=suggest-override]
   25 |  virtual PropertyInfo _gen_argument_type_info(int p_argument) const {
      |                       ^~~~~~~~~~~~~~~~~~~~~~~
./core/method_bind.gen.inc:32:17: error: 'virtual String MethodBind0::get_instance_class() const' can be marked override [-Werror=suggest-override]
   32 |  virtual String get_instance_class() const {
      |                 ^~~~~~~~~~~~~~~~~~
./core/method_bind.gen.inc:36:18: error: 'virtual Variant MethodBind0::call(Object*, const Variant**, int, Callable::CallError&)' can be marked override [-Werror=suggest-override]
   36 |  virtual Variant call(Object* p_object,const Variant** p_args,int p_arg_count, Callable::CallError& r_error) {
      |                  ^~~~
./core/method_bind.gen.inc:64:15: error: 'virtual void MethodBind0::ptrcall(Object*, const void**, void*)' can be marked override [-Werror=suggest-override]
   64 |  virtual void ptrcall(Object*p_object,const void** p_args,void *r_ret) {
      |               ^~~~~~~
./core/method_bind.gen.inc:368:24: error: 'virtual Variant::Type MethodBind0C::_gen_argument_type(int) const' can be marked override [-Werror=suggest-override]
  368 |  virtual Variant::Type _gen_argument_type(int p_arg) const { return _get_argument_type(p_arg); }
      |                        ^~~~~~~~~~~~~~~~~~
./core/method_bind.gen.inc:369:34: error: 'virtual GodotTypeInfo::Metadata MethodBind0C::get_argument_meta(int) const' can be marked override [-Werror=suggest-override]
  369 |  virtual GodotTypeInfo::Metadata get_argument_meta(int p_arg) const {
      |                                  ^~~~~~~~~~~~~~~~~
./core/method_bind.gen.inc:381:23: error: 'virtual PropertyInfo MethodBind0C::_gen_argument_type_info(int) const' can be marked override [-Werror=suggest-override]
  381 |  virtual PropertyInfo _gen_argument_type_info(int p_argument) const {
      |                       ^~~~~~~~~~~~~~~~~~~~~~~
./core/method_bind.gen.inc:388:17: error: 'virtual String MethodBind0C::get_instance_class() const' can be marked override [-Werror=suggest-override]
  388 |  virtual String get_instance_class() const {
      |                 ^~~~~~~~~~~~~~~~~~
./core/method_bind.gen.inc:392:18: error: 'virtual Variant MethodBind0C::call(Object*, const Variant**, int, Callable::CallError&)' can be marked override [-Werror=suggest-override]
  392 |  virtual Variant call(Object* p_object,const Variant** p_args,int p_arg_count, Callable::CallError& r_error) {
      |                  ^~~~
./core/method_bind.gen.inc:420:15: error: 'virtual void MethodBind0C::ptrcall(Object*, const void**, void*)' can be marked override [-Werror=suggest-override]
  420 |  virtual void ptrcall(Object*p_object,const void** p_args,void *r_ret) {
      |               ^~~~~~~
In file included from ./core/class_db.h:44,
                 from ./core/resource.h:34,
                 from ./core/input/input_event.h:36,
                 from ./core/input/input.h:34,
                 from platform/linuxbsd/os_linuxbsd.h:34,
                 from platform/linuxbsd/godot_linuxbsd.cpp:37:
./core/callable_method_pointer.h:58:17: error: 'virtual String CallableCustomMethodPointerBase::get_as_text() const' can be marked override [-Werror=suggest-override]
   58 |  virtual String get_as_text() const {
      |                 ^~~~~~~~~~~
./core/callable_method_pointer.h:66:27: error: 'virtual bool (* CallableCustomMethodPointerBase::get_compare_equal_func() const)(const CallableCustom*, const CallableCustom*)' can be marked override [-Werror=suggest-override]
   66 |  virtual CompareEqualFunc get_compare_equal_func() const;
      |                           ^~~~~~~~~~~~~~~~~~~~~~
./core/callable_method_pointer.h:67:26: error: 'virtual bool (* CallableCustomMethodPointerBase::get_compare_less_func() const)(const CallableCustom*, const CallableCustom*)' can be marked override [-Werror=suggest-override]
   67 |  virtual CompareLessFunc get_compare_less_func() const;
      |                          ^~~~~~~~~~~~~~~~~~~~~
./core/callable_method_pointer.h:69:19: error: 'virtual uint32_t CallableCustomMethodPointerBase::hash() const' can be marked override [-Werror=suggest-override]
   69 |  virtual uint32_t hash() const;
      |                   ^~~~
In file included from ./core/os/main_loop.h:36,
                 from ./core/engine.h:35,
                 from ./core/os/os.h:34,
                 from ./servers/audio_server.h:36,
                 from ./drivers/alsa/audio_driver_alsa.h:38,
                 from platform/linuxbsd/os_linuxbsd.h:36,
                 from platform/linuxbsd/godot_linuxbsd.cpp:37:
./core/script_language.h:416:15: error: 'virtual bool PlaceHolderScriptInstance::set(const StringName&, const Variant&)' can be marked override [-Werror=suggest-override]
  416 |  virtual bool set(const StringName &p_name, const Variant &p_value);
      |               ^~~
./core/script_language.h:417:15: error: 'virtual bool PlaceHolderScriptInstance::get(const StringName&, Variant&) const' can be marked override [-Werror=suggest-override]
  417 |  virtual bool get(const StringName &p_name, Variant &r_ret) const;
      |               ^~~
./core/script_language.h:418:15: error: 'virtual void PlaceHolderScriptInstance::get_property_list(List<PropertyInfo>*) const' can be marked override [-Werror=suggest-override]
  418 |  virtual void get_property_list(List<PropertyInfo> *p_properties) const;
      |               ^~~~~~~~~~~~~~~~~
./core/script_language.h:419:24: error: 'virtual Variant::Type PlaceHolderScriptInstance::get_property_type(const StringName&, bool*) const' can be marked override [-Werror=suggest-override]
  419 |  virtual Variant::Type get_property_type(const StringName &p_name, bool *r_is_valid = nullptr) const;
      |                        ^~~~~~~~~~~~~~~~~
./core/script_language.h:421:15: error: 'virtual void PlaceHolderScriptInstance::get_method_list(List<MethodInfo>*) const' can be marked override [-Werror=suggest-override]
  421 |  virtual void get_method_list(List<MethodInfo> *p_list) const;
      |               ^~~~~~~~~~~~~~~
./core/script_language.h:422:15: error: 'virtual bool PlaceHolderScriptInstance::has_method(const StringName&) const' can be marked override [-Werror=suggest-override]
  422 |  virtual bool has_method(const StringName &p_method) const;
      |               ^~~~~~~~~~
./core/script_language.h:423:18: error: 'virtual Variant PlaceHolderScriptInstance::call(const StringName&, const Variant&, const Variant&, const Variant&, const Variant&, const Variant&)' can be marked override [-Werror=suggest-override]
  423 |  virtual Variant call(const StringName &p_method, VARIANT_ARG_LIST) { return Variant(); }
      |                  ^~~~
./core/script_language.h:424:18: error: 'virtual Variant PlaceHolderScriptInstance::call(const StringName&, const Variant**, int, Callable::CallError&)' can be marked override [-Werror=suggest-override]
  424 |  virtual Variant call(const StringName &p_method, const Variant **p_args, int p_argcount, Callable::CallError &r_error) {
      |                  ^~~~
./core/script_language.h:428:15: error: 'virtual void PlaceHolderScriptInstance::notification(int)' can be marked override [-Werror=suggest-override]
  428 |  virtual void notification(int p_notification) {}
      |               ^~~~~~~~~~~~
./core/script_language.h:430:22: error: 'virtual Ref<Script> PlaceHolderScriptInstance::get_script() const' can be marked override [-Werror=suggest-override]
  430 |  virtual Ref<Script> get_script() const { return script; }
      |                      ^~~~~~~~~~
./core/script_language.h:432:26: error: 'virtual ScriptLanguage* PlaceHolderScriptInstance::get_language()' can be marked override [-Werror=suggest-override]
  432 |  virtual ScriptLanguage *get_language() { return language; }
      |                          ^~~~~~~~~~~~
./core/script_language.h:434:10: error: 'virtual Object* PlaceHolderScriptInstance::get_owner()' can be marked override [-Werror=suggest-override]
  434 |  Object *get_owner() { return owner; }
      |          ^~~~~~~~~
./core/script_language.h:438:15: error: 'virtual bool PlaceHolderScriptInstance::is_placeholder() const' can be marked override [-Werror=suggest-override]
  438 |  virtual bool is_placeholder() const { return true; }
      |               ^~~~~~~~~~~~~~
./core/script_language.h:440:15: error: 'virtual void PlaceHolderScriptInstance::property_set_fallback(const StringName&, const Variant&, bool*)' can be marked override [-Werror=suggest-override]
  440 |  virtual void property_set_fallback(const StringName &p_name, const Variant &p_value, bool *r_valid = nullptr);
      |               ^~~~~~~~~~~~~~~~~~~~~
./core/script_language.h:441:18: error: 'virtual Variant PlaceHolderScriptInstance::property_get_fallback(const StringName&, bool*)' can be marked override [-Werror=suggest-override]
  441 |  virtual Variant property_get_fallback(const StringName &p_name, bool *r_valid = nullptr);
      |                  ^~~~~~~~~~~~~~~~~~~~~
./core/script_language.h:443:32: error: 'virtual Vector<ScriptNetData> PlaceHolderScriptInstance::get_rpc_methods() const' can be marked override [-Werror=suggest-override]
  443 |  virtual Vector<ScriptNetData> get_rpc_methods() const { return Vector<ScriptNetData>(); }
      |                                ^~~~~~~~~~~~~~~
./core/script_language.h:444:19: error: 'virtual uint16_t PlaceHolderScriptInstance::get_rpc_method_id(const StringName&) const' can be marked override [-Werror=suggest-override]
  444 |  virtual uint16_t get_rpc_method_id(const StringName &p_method) const;
      |                   ^~~~~~~~~~~~~~~~~
./core/script_language.h:445:21: error: 'virtual StringName PlaceHolderScriptInstance::get_rpc_method(uint16_t) const' can be marked override [-Werror=suggest-override]
  445 |  virtual StringName get_rpc_method(uint16_t p_id) const { return StringName(); }
      |                     ^~~~~~~~~~~~~~
./core/script_language.h:446:34: error: 'virtual MultiplayerAPI::RPCMode PlaceHolderScriptInstance::get_rpc_mode_by_id(uint16_t) const' can be marked override [-Werror=suggest-override]
  446 |  virtual MultiplayerAPI::RPCMode get_rpc_mode_by_id(uint16_t p_id) const { return MultiplayerAPI::RPC_MODE_DISABLED; }
      |                                  ^~~~~~~~~~~~~~~~~~
./core/script_language.h:447:34: error: 'virtual MultiplayerAPI::RPCMode PlaceHolderScriptInstance::get_rpc_mode(const StringName&) const' can be marked override [-Werror=suggest-override]
  447 |  virtual MultiplayerAPI::RPCMode get_rpc_mode(const StringName &p_method) const { return MultiplayerAPI::RPC_MODE_DISABLED; }
      |                                  ^~~~~~~~~~~~
./core/script_language.h:449:32: error: 'virtual Vector<ScriptNetData> PlaceHolderScriptInstance::get_rset_properties() const' can be marked override [-Werror=suggest-override]
  449 |  virtual Vector<ScriptNetData> get_rset_properties() const { return Vector<ScriptNetData>(); }
      |                                ^~~~~~~~~~~~~~~~~~~
./core/script_language.h:450:19: error: 'virtual uint16_t PlaceHolderScriptInstance::get_rset_property_id(const StringName&) const' can be marked override [-Werror=suggest-override]
  450 |  virtual uint16_t get_rset_property_id(const StringName &p_variable) const;
      |                   ^~~~~~~~~~~~~~~~~~~~
./core/script_language.h:451:21: error: 'virtual StringName PlaceHolderScriptInstance::get_rset_property(uint16_t) const' can be marked override [-Werror=suggest-override]
  451 |  virtual StringName get_rset_property(uint16_t p_id) const { return StringName(); }
      |                     ^~~~~~~~~~~~~~~~~
./core/script_language.h:452:34: error: 'virtual MultiplayerAPI::RPCMode PlaceHolderScriptInstance::get_rset_mode_by_id(uint16_t) const' can be marked override [-Werror=suggest-override]
  452 |  virtual MultiplayerAPI::RPCMode get_rset_mode_by_id(uint16_t p_id) const { return MultiplayerAPI::RPC_MODE_DISABLED; }
      |                                  ^~~~~~~~~~~~~~~~~~~
./core/script_language.h:453:34: error: 'virtual MultiplayerAPI::RPCMode PlaceHolderScriptInstance::get_rset_mode(const StringName&) const' can be marked override [-Werror=suggest-override]
  453 |  virtual MultiplayerAPI::RPCMode get_rset_mode(const StringName &p_variable) const { return MultiplayerAPI::RPC_MODE_DISABLED; }
      |                                  ^~~~~~~~~~~~~
In file included from ./core/os/os.h:36,
                 from ./servers/audio_server.h:36,
                 from ./drivers/alsa/audio_driver_alsa.h:38,
                 from platform/linuxbsd/os_linuxbsd.h:36,
                 from platform/linuxbsd/godot_linuxbsd.cpp:37:
./core/io/logger.h:66:15: error: 'virtual void StdLogger::logv(const char*, __va_list_tag*, bool)' can be marked override [-Werror=suggest-override]
   66 |  virtual void logv(const char *p_format, va_list p_list, bool p_err) _PRINTF_FORMAT_ATTRIBUTE_2_0;
      |               ^~~~
./core/io/logger.h:90:15: error: 'virtual void RotatedFileLogger::logv(const char*, __va_list_tag*, bool)' can be marked override [-Werror=suggest-override]
   90 |  virtual void logv(const char *p_format, va_list p_list, bool p_err) _PRINTF_FORMAT_ATTRIBUTE_2_0;
      |               ^~~~
./core/io/logger.h:101:15: error: 'virtual void CompositeLogger::logv(const char*, __va_list_tag*, bool)' can be marked override [-Werror=suggest-override]
  101 |  virtual void logv(const char *p_format, va_list p_list, bool p_err) _PRINTF_FORMAT_ATTRIBUTE_2_0;
      |               ^~~~
./core/io/logger.h:102:15: error: 'virtual void CompositeLogger::log_error(const char*, const char*, int, const char*, const char*, Logger::ErrorType)' can be marked override [-Werror=suggest-override]
  102 |  virtual void log_error(const char *p_function, const char *p_file, int p_line, const char *p_code, const char *p_rationale, ErrorType p_type = ERR_ERROR);
      |               ^~~~~~~~~
In file included from platform/linuxbsd/os_linuxbsd.h:36,
                 from platform/linuxbsd/godot_linuxbsd.cpp:37:
./drivers/alsa/audio_driver_alsa.h:72:14: error: 'virtual const char* AudioDriverALSA::get_name() const' can be marked override [-Werror=suggest-override]
   72 |  const char *get_name() const {
      |              ^~~~~~~~
./drivers/alsa/audio_driver_alsa.h:76:16: error: 'virtual Error AudioDriverALSA::init()' can be marked override [-Werror=suggest-override]
   76 |  virtual Error init();
      |                ^~~~
./drivers/alsa/audio_driver_alsa.h:77:15: error: 'virtual void AudioDriverALSA::start()' can be marked override [-Werror=suggest-override]
   77 |  virtual void start();
      |               ^~~~~
./drivers/alsa/audio_driver_alsa.h:78:14: error: 'virtual int AudioDriverALSA::get_mix_rate() const' can be marked override [-Werror=suggest-override]
   78 |  virtual int get_mix_rate() const;
      |              ^~~~~~~~~~~~
./drivers/alsa/audio_driver_alsa.h:79:22: error: 'virtual AudioDriver::SpeakerMode AudioDriverALSA::get_speaker_mode() const' can be marked override [-Werror=suggest-override]
   79 |  virtual SpeakerMode get_speaker_mode() const;
      |                      ^~~~~~~~~~~~~~~~
./drivers/alsa/audio_driver_alsa.h:80:16: error: 'virtual Array AudioDriverALSA::get_device_list()' can be marked override [-Werror=suggest-override]
   80 |  virtual Array get_device_list();
      |                ^~~~~~~~~~~~~~~
./drivers/alsa/audio_driver_alsa.h:81:17: error: 'virtual String AudioDriverALSA::get_device()' can be marked override [-Werror=suggest-override]
   81 |  virtual String get_device();
      |                 ^~~~~~~~~~
./drivers/alsa/audio_driver_alsa.h:82:15: error: 'virtual void AudioDriverALSA::set_device(String)' can be marked override [-Werror=suggest-override]
   82 |  virtual void set_device(String device);
      |               ^~~~~~~~~~
./drivers/alsa/audio_driver_alsa.h:83:15: error: 'virtual void AudioDriverALSA::lock()' can be marked override [-Werror=suggest-override]
   83 |  virtual void lock();
      |               ^~~~
./drivers/alsa/audio_driver_alsa.h:84:15: error: 'virtual void AudioDriverALSA::unlock()' can be marked override [-Werror=suggest-override]
   84 |  virtual void unlock();
      |               ^~~~~~
./drivers/alsa/audio_driver_alsa.h:85:15: error: 'virtual void AudioDriverALSA::finish()' can be marked override [-Werror=suggest-override]
   85 |  virtual void finish();
      |               ^~~~~~
In file included from platform/linuxbsd/os_linuxbsd.h:37,
                 from platform/linuxbsd/godot_linuxbsd.cpp:37:
./drivers/alsamidi/midi_driver_alsamidi.h:58:16: error: 'virtual Error MIDIDriverALSAMidi::open()' can be marked override [-Werror=suggest-override]
   58 |  virtual Error open();
      |                ^~~~
./drivers/alsamidi/midi_driver_alsamidi.h:59:15: error: 'virtual void MIDIDriverALSAMidi::close()' can be marked override [-Werror=suggest-override]
   59 |  virtual void close();
      |               ^~~~~
./drivers/alsamidi/midi_driver_alsamidi.h:61:28: error: 'virtual PackedStringArray MIDIDriverALSAMidi::get_connected_inputs()' can be marked override [-Werror=suggest-override]
   61 |  virtual PackedStringArray get_connected_inputs();
      |                            ^~~~~~~~~~~~~~~~~~~~
In file included from platform/linuxbsd/os_linuxbsd.h:38,
                 from platform/linuxbsd/godot_linuxbsd.cpp:37:
./drivers/pulseaudio/audio_driver_pulseaudio.h:97:14: error: 'virtual const char* AudioDriverPulseAudio::get_name() const' can be marked override [-Werror=suggest-override]
   97 |  const char *get_name() const {
      |              ^~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:101:16: error: 'virtual Error AudioDriverPulseAudio::init()' can be marked override [-Werror=suggest-override]
  101 |  virtual Error init();
      |                ^~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:102:15: error: 'virtual void AudioDriverPulseAudio::start()' can be marked override [-Werror=suggest-override]
  102 |  virtual void start();
      |               ^~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:103:14: error: 'virtual int AudioDriverPulseAudio::get_mix_rate() const' can be marked override [-Werror=suggest-override]
  103 |  virtual int get_mix_rate() const;
      |              ^~~~~~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:104:22: error: 'virtual AudioDriver::SpeakerMode AudioDriverPulseAudio::get_speaker_mode() const' can be marked override [-Werror=suggest-override]
  104 |  virtual SpeakerMode get_speaker_mode() const;
      |                      ^~~~~~~~~~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:106:16: error: 'virtual Array AudioDriverPulseAudio::get_device_list()' can be marked override [-Werror=suggest-override]
  106 |  virtual Array get_device_list();
      |                ^~~~~~~~~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:107:17: error: 'virtual String AudioDriverPulseAudio::get_device()' can be marked override [-Werror=suggest-override]
  107 |  virtual String get_device();
      |                 ^~~~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:108:15: error: 'virtual void AudioDriverPulseAudio::set_device(String)' can be marked override [-Werror=suggest-override]
  108 |  virtual void set_device(String device);
      |               ^~~~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:110:16: error: 'virtual Array AudioDriverPulseAudio::capture_get_device_list()' can be marked override [-Werror=suggest-override]
  110 |  virtual Array capture_get_device_list();
      |                ^~~~~~~~~~~~~~~~~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:111:15: error: 'virtual void AudioDriverPulseAudio::capture_set_device(const String&)' can be marked override [-Werror=suggest-override]
  111 |  virtual void capture_set_device(const String &p_name);
      |               ^~~~~~~~~~~~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:112:17: error: 'virtual String AudioDriverPulseAudio::capture_get_device()' can be marked override [-Werror=suggest-override]
  112 |  virtual String capture_get_device();
      |                 ^~~~~~~~~~~~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:114:15: error: 'virtual void AudioDriverPulseAudio::lock()' can be marked override [-Werror=suggest-override]
  114 |  virtual void lock();
      |               ^~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:115:15: error: 'virtual void AudioDriverPulseAudio::unlock()' can be marked override [-Werror=suggest-override]
  115 |  virtual void unlock();
      |               ^~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:116:15: error: 'virtual void AudioDriverPulseAudio::finish()' can be marked override [-Werror=suggest-override]
  116 |  virtual void finish();
      |               ^~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:118:16: error: 'virtual float AudioDriverPulseAudio::get_latency()' can be marked override [-Werror=suggest-override]
  118 |  virtual float get_latency();
      |                ^~~~~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:120:16: error: 'virtual Error AudioDriverPulseAudio::capture_start()' can be marked override [-Werror=suggest-override]
  120 |  virtual Error capture_start();
      |                ^~~~~~~~~~~~~
./drivers/pulseaudio/audio_driver_pulseaudio.h:121:16: error: 'virtual Error AudioDriverPulseAudio::capture_stop()' can be marked override [-Werror=suggest-override]
  121 |  virtual Error capture_stop();
      |                ^~~~~~~~~~~~
In file included from platform/linuxbsd/os_linuxbsd.h:39,
                 from platform/linuxbsd/godot_linuxbsd.cpp:37:
./drivers/unix/os_unix.h:48:15: error: 'virtual void OS_Unix::finalize_core()' can be marked override [-Werror=suggest-override]
   48 |  virtual void finalize_core();
      |               ^~~~~~~~~~~~~
./drivers/unix/os_unix.h:56:17: error: 'virtual String OS_Unix::get_stdin_string(bool)' can be marked override [-Werror=suggest-override]
   56 |  virtual String get_stdin_string(bool p_block);
      |                 ^~~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:68:16: error: 'virtual Error OS_Unix::open_dynamic_library(String, void*&, bool)' can be marked override [-Werror=suggest-override]
   68 |  virtual Error open_dynamic_library(const String p_path, void *&p_library_handle, bool p_also_set_library_path = false);
      |                ^~~~~~~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:69:16: error: 'virtual Error OS_Unix::close_dynamic_library(void*)' can be marked override [-Werror=suggest-override]
   69 |  virtual Error close_dynamic_library(void *p_library_handle);
      |                ^~~~~~~~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:70:16: error: 'virtual Error OS_Unix::get_dynamic_library_symbol_handle(void*, String, void*&, bool)' can be marked override [-Werror=suggest-override]
   70 |  virtual Error get_dynamic_library_symbol_handle(void *p_library_handle, const String p_name, void *&p_symbol_handle, bool p_optional = false);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from platform/linuxbsd/os_linuxbsd.h:39,
                 from platform/linuxbsd/godot_linuxbsd.cpp:37:
./drivers/unix/os_unix.h:72:16: error: 'virtual Error OS_Unix::set_cwd(const String&)' can be marked override [-Werror=suggest-override]
   72 |  virtual Error set_cwd(const String &p_cwd);
      |                ^~~~~~~
./drivers/unix/os_unix.h:74:17: error: 'virtual String OS_Unix::get_name() const' can be marked override [-Werror=suggest-override]
   74 |  virtual String get_name() const;
      |                 ^~~~~~~~
./drivers/unix/os_unix.h:76:15: error: 'virtual OS::Date OS_Unix::get_date(bool) const' can be marked override [-Werror=suggest-override]
   76 |  virtual Date get_date(bool utc) const;
      |               ^~~~~~~~
./drivers/unix/os_unix.h:77:15: error: 'virtual OS::Time OS_Unix::get_time(bool) const' can be marked override [-Werror=suggest-override]
   77 |  virtual Time get_time(bool utc) const;
      |               ^~~~~~~~
./drivers/unix/os_unix.h:78:23: error: 'virtual OS::TimeZoneInfo OS_Unix::get_time_zone_info() const' can be marked override [-Werror=suggest-override]
   78 |  virtual TimeZoneInfo get_time_zone_info() const;
      |                       ^~~~~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:80:17: error: 'virtual double OS_Unix::get_unix_time() const' can be marked override [-Werror=suggest-override]
   80 |  virtual double get_unix_time() const;
      |                 ^~~~~~~~~~~~~
./drivers/unix/os_unix.h:82:15: error: 'virtual void OS_Unix::delay_usec(uint32_t) const' can be marked override [-Werror=suggest-override]
   82 |  virtual void delay_usec(uint32_t p_usec) const;
      |               ^~~~~~~~~~
./drivers/unix/os_unix.h:83:19: error: 'virtual uint64_t OS_Unix::get_ticks_usec() const' can be marked override [-Werror=suggest-override]
   83 |  virtual uint64_t get_ticks_usec() const;
      |                   ^~~~~~~~~~~~~~
./drivers/unix/os_unix.h:85:16: error: 'virtual Error OS_Unix::execute(const String&, const List<String>&, bool, OS::ProcessID*, String*, int*, bool, Mutex*)' can be marked override [-Werror=suggest-override]
   85 |  virtual Error execute(const String &p_path, const List<String> &p_arguments, bool p_blocking = true, ProcessID *r_child_id = nullptr, String *r_pipe = nullptr, int *r_exitcode = nullptr, bool read_stderr = false, Mutex *p_pipe_mutex = nullptr);
      |                ^~~~~~~
./drivers/unix/os_unix.h:86:16: error: 'virtual Error OS_Unix::kill(const ProcessID&)' can be marked override [-Werror=suggest-override]
   86 |  virtual Error kill(const ProcessID &p_pid);
      |                ^~~~
./drivers/unix/os_unix.h:87:14: error: 'virtual int OS_Unix::get_process_id() const' can be marked override [-Werror=suggest-override]
   87 |  virtual int get_process_id() const;
      |              ^~~~~~~~~~~~~~
./drivers/unix/os_unix.h:89:15: error: 'virtual bool OS_Unix::has_environment(const String&) const' can be marked override [-Werror=suggest-override]
   89 |  virtual bool has_environment(const String &p_var) const;
      |               ^~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:90:17: error: 'virtual String OS_Unix::get_environment(const String&) const' can be marked override [-Werror=suggest-override]
   90 |  virtual String get_environment(const String &p_var) const;
      |                 ^~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:91:15: error: 'virtual bool OS_Unix::set_environment(const String&, const String&) const' can be marked override [-Werror=suggest-override]
   91 |  virtual bool set_environment(const String &p_var, const String &p_value) const;
      |               ^~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:92:17: error: 'virtual String OS_Unix::get_locale() const' can be marked override [-Werror=suggest-override]
   92 |  virtual String get_locale() const;
      |                 ^~~~~~~~~~
./drivers/unix/os_unix.h:94:14: error: 'virtual int OS_Unix::get_processor_count() const' can be marked override [-Werror=suggest-override]
   94 |  virtual int get_processor_count() const;
      |              ^~~~~~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:96:15: error: 'virtual void OS_Unix::debug_break()' can be marked override [-Werror=suggest-override]
   96 |  virtual void debug_break();
      |               ^~~~~~~~~~~
./drivers/unix/os_unix.h:97:15: error: 'virtual void OS_Unix::initialize_debugging()' can be marked override [-Werror=suggest-override]
   97 |  virtual void initialize_debugging();
      |               ^~~~~~~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:99:17: error: 'virtual String OS_Unix::get_executable_path() const' can be marked override [-Werror=suggest-override]
   99 |  virtual String get_executable_path() const;
      |                 ^~~~~~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:100:17: error: 'virtual String OS_Unix::get_user_data_dir() const' can be marked override [-Werror=suggest-override]
  100 |  virtual String get_user_data_dir() const;
      |                 ^~~~~~~~~~~~~~~~~
./drivers/unix/os_unix.h:105:15: error: 'virtual void UnixTerminalLogger::log_error(const char*, const char*, int, const char*, const char*, Logger::ErrorType)' can be marked override [-Werror=suggest-override]
  105 |  virtual void log_error(const char *p_function, const char *p_file, int p_line, const char *p_code, const char *p_rationale, ErrorType p_type = ERR_ERROR);
      |               ^~~~~~~~~
In file included from ./servers/rendering_server.h:43,
                 from ./servers/rendering/rasterizer.h:37,
                 from platform/linuxbsd/os_linuxbsd.h:42,
                 from platform/linuxbsd/godot_linuxbsd.cpp:37:
./servers/rendering/shader_language.h:374:20: error: 'virtual ShaderLanguage::DataType ShaderLanguage::OperatorNode::get_datatype() const' can be marked override [-Werror=suggest-override]
  374 |   virtual DataType get_datatype() const { return return_cache; }
      |                    ^~~~~~~~~~~~
./servers/rendering/shader_language.h:375:18: error: 'virtual String ShaderLanguage::OperatorNode::get_datatype_name() const' can be marked override [-Werror=suggest-override]
  375 |   virtual String get_datatype_name() const { return String(struct_name); }
      |                  ^~~~~~~~~~~~~~~~~
./servers/rendering/shader_language.h:385:20: error: 'virtual ShaderLanguage::DataType ShaderLanguage::VariableNode::get_datatype() const' can be marked override [-Werror=suggest-override]
  385 |   virtual DataType get_datatype() const { return datatype_cache; }
      |                    ^~~~~~~~~~~~
./servers/rendering/shader_language.h:386:18: error: 'virtual String ShaderLanguage::VariableNode::get_datatype_name() const' can be marked override [-Werror=suggest-override]
  386 |   virtual String get_datatype_name() const { return String(struct_name); }
      |                  ^~~~~~~~~~~~~~~~~
./servers/rendering/shader_language.h:405:20: error: 'virtual ShaderLanguage::DataType ShaderLanguage::VariableDeclarationNode::get_datatype() const' can be marked override [-Werror=suggest-override]
  405 |   virtual DataType get_datatype() const { return datatype; }
      |                    ^~~~~~~~~~~~
./servers/rendering/shader_language.h:419:20: error: 'virtual ShaderLanguage::DataType ShaderLanguage::ArrayNode::get_datatype() const' can be marked override [-Werror=suggest-override]
  419 |   virtual DataType get_datatype() const { return datatype_cache; }
      |                    ^~~~~~~~~~~~
./servers/rendering/shader_language.h:420:18: error: 'virtual String ShaderLanguage::ArrayNode::get_datatype_name() const' can be marked override [-Werror=suggest-override]
  420 |   virtual String get_datatype_name() const { return String(struct_name); }
      |                  ^~~~~~~~~~~~~~~~~
./servers/rendering/shader_language.h:448:20: error: 'virtual ShaderLanguage::DataType ShaderLanguage::ArrayDeclarationNode::get_datatype() const' can be marked override [-Werror=suggest-override]
  448 |   virtual DataType get_datatype() const { return datatype; }
      |                    ^~~~~~~~~~~~
./servers/rendering/shader_language.h:468:20: error: 'virtual ShaderLanguage::DataType ShaderLanguage::ConstantNode::get_datatype() const' can be marked override [-Werror=suggest-override]
  468 |   virtual DataType get_datatype() const { return datatype; }
      |                    ^~~~~~~~~~~~
./servers/rendering/shader_language.h:469:18: error: 'virtual String ShaderLanguage::ConstantNode::get_datatype_name() const' can be marked override [-Werror=suggest-override]
  469 |   virtual String get_datatype_name() const { return struct_name; }
      |                  ^~~~~~~~~~~~~~~~~
./servers/rendering/shader_language.h:531:20: error: 'virtual ShaderLanguage::DataType ShaderLanguage::MemberNode::get_datatype() const' can be marked override [-Werror=suggest-override]
  531 |   virtual DataType get_datatype() const { return datatype; }
      |                    ^~~~~~~~~~~~
./servers/rendering/shader_language.h:532:18: error: 'virtual String ShaderLanguage::MemberNode::get_datatype_name() const' can be marked override [-Werror=suggest-override]
  532 |   virtual String get_datatype_name() const { return String(struct_name); }
      |                  ^~~~~~~~~~~~~~~~~
In file included from platform/linuxbsd/godot_linuxbsd.cpp:37:
platform/linuxbsd/os_linuxbsd.h:46:15: error: 'virtual void OS_LinuxBSD::delete_main_loop()' can be marked override [-Werror=suggest-override]
   46 |  virtual void delete_main_loop();
      |               ^~~~~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:71:15: error: 'virtual void OS_LinuxBSD::initialize()' can be marked override [-Werror=suggest-override]
   71 |  virtual void initialize();
      |               ^~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:72:15: error: 'virtual void OS_LinuxBSD::finalize()' can be marked override [-Werror=suggest-override]
   72 |  virtual void finalize();
      |               ^~~~~~~~
platform/linuxbsd/os_linuxbsd.h:74:15: error: 'virtual void OS_LinuxBSD::initialize_joypads()' can be marked override [-Werror=suggest-override]
   74 |  virtual void initialize_joypads();
      |               ^~~~~~~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:76:15: error: 'virtual void OS_LinuxBSD::set_main_loop(MainLoop*)' can be marked override [-Werror=suggest-override]
   76 |  virtual void set_main_loop(MainLoop *p_main_loop);
      |               ^~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:79:17: error: 'virtual String OS_LinuxBSD::get_name() const' can be marked override [-Werror=suggest-override]
   79 |  virtual String get_name() const;
      |                 ^~~~~~~~
platform/linuxbsd/os_linuxbsd.h:81:20: error: 'virtual MainLoop* OS_LinuxBSD::get_main_loop() const' can be marked override [-Werror=suggest-override]
   81 |  virtual MainLoop *get_main_loop() const;
      |                    ^~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:83:17: error: 'virtual String OS_LinuxBSD::get_config_path() const' can be marked override [-Werror=suggest-override]
   83 |  virtual String get_config_path() const;
      |                 ^~~~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:84:17: error: 'virtual String OS_LinuxBSD::get_data_path() const' can be marked override [-Werror=suggest-override]
   84 |  virtual String get_data_path() const;
      |                 ^~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:85:17: error: 'virtual String OS_LinuxBSD::get_cache_path() const' can be marked override [-Werror=suggest-override]
   85 |  virtual String get_cache_path() const;
      |                 ^~~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:87:17: error: 'virtual String OS_LinuxBSD::get_system_dir(OS::SystemDir) const' can be marked override [-Werror=suggest-override]
   87 |  virtual String get_system_dir(SystemDir p_dir) const;
      |                 ^~~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:89:16: error: 'virtual Error OS_LinuxBSD::shell_open(String)' can be marked override [-Werror=suggest-override]
   89 |  virtual Error shell_open(String p_uri);
      |                ^~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:91:17: error: 'virtual String OS_LinuxBSD::get_unique_id() const' can be marked override [-Werror=suggest-override]
   91 |  virtual String get_unique_id() const;
      |                 ^~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:93:15: error: 'virtual bool OS_LinuxBSD::_check_internal_feature_support(const String&)' can be marked override [-Werror=suggest-override]
   93 |  virtual bool _check_internal_feature_support(const String &p_feature);
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:97:7: error: 'virtual void OS_LinuxBSD::disable_crash_handler()' can be marked override [-Werror=suggest-override]
   97 |  void disable_crash_handler();
      |       ^~~~~~~~~~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:98:7: error: 'virtual bool OS_LinuxBSD::is_disable_crash_handler() const' can be marked override [-Werror=suggest-override]
   98 |  bool is_disable_crash_handler() const;
      |       ^~~~~~~~~~~~~~~~~~~~~~~~
platform/linuxbsd/os_linuxbsd.h:100:16: error: 'virtual Error OS_LinuxBSD::move_to_trash(const String&)' can be marked override [-Werror=suggest-override]
  100 |  virtual Error move_to_trash(const String &p_path);
      |                ^~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

So we need to fix those before we can enable the warning.

--- a/SConstruct
+++ b/SConstruct
@@ -414,7 +414,7 @@ if selected_platform in platform_list:
             Exit(255)
 
     # Configure compiler warnings
-    if env.msvc:
+    if env.msvc:  # MSVC
         # Truncations, narrowing conversions, signed/unsigned comparisons...
         disable_nonessential_warnings = ["/wd4267", "/wd4244", "/wd4305", "/wd4018", "/wd4800"]
         if env["warnings"] == "extra":
@@ -427,21 +427,17 @@ if selected_platform in platform_list:
             env.Append(CCFLAGS=["/w"])
         # Set exception handling model to avoid warnings caused by Windows system headers.
         env.Append(CCFLAGS=["/EHsc"])
+
         if env["werror"]:
             env.Append(CCFLAGS=["/WX"])
-        # Force to use Unicode encoding
-        env.Append(MSVC_FLAGS=["/utf8"])
-    else:  # Rest of the world
-        shadow_local_warning = []
-        all_plus_warnings = ["-Wwrite-strings"]
+    else:  # GCC, Clang
+        gcc_common_warnings = []
 
         if methods.using_gcc(env):
-            env.Append(CCFLAGS=["-Wno-misleading-indentation"])
-            if cc_version_major >= 7:
-                shadow_local_warning = ["-Wshadow-local"]
+            gcc_common_warnings += ["-Wshadow-local", "-Wno-misleading-indentation"]
 
         if env["warnings"] == "extra":
-            env.Append(CCFLAGS=["-Wall", "-Wextra", "-Wno-unused-parameter"] + all_plus_warnings + shadow_local_warning)
+            env.Append(CCFLAGS=["-Wall", "-Wextra", "-Wwrite-strings", "-Wno-unused-parameter"] + gcc_common_warnings)
             env.Append(CXXFLAGS=["-Wctor-dtor-privacy", "-Wnon-virtual-dtor"])
             if methods.using_gcc(env):
                 env.Append(
@@ -457,14 +453,15 @@ if selected_platform in platform_list:
                 env.Append(CXXFLAGS=["-Wplacement-new=1"])
                 if cc_version_major >= 9:
                     env.Append(CCFLAGS=["-Wattribute-alias=2"])
-            if methods.using_clang(env):
+            elif methods.using_clang(env):
                 env.Append(CCFLAGS=["-Wimplicit-fallthrough"])
         elif env["warnings"] == "all":
-            env.Append(CCFLAGS=["-Wall"] + shadow_local_warning)
+            env.Append(CCFLAGS=["-Wall"] + gcc_common_warnings)
         elif env["warnings"] == "moderate":
-            env.Append(CCFLAGS=["-Wall", "-Wno-unused"] + shadow_local_warning)
+            env.Append(CCFLAGS=["-Wall", "-Wno-unused"] + gcc_common_warnings)
         else:  # 'no'
             env.Append(CCFLAGS=["-w"])
+
         if env["werror"]:
             env.Append(CCFLAGS=["-Werror"])
             # FIXME: Temporary workaround after the Vulkan merge, remove once warnings are fixed.

--- a/platform/uwp/detect.py
+++ b/platform/uwp/detect.py
@@ -80,6 +80,9 @@ def configure(env):
     env["ENV"] = os.environ
     vc_base_path = os.environ["VCTOOLSINSTALLDIR"] if "VCTOOLSINSTALLDIR" in os.environ else os.environ["VCINSTALLDIR"]
 
+    # Force to use Unicode encoding
+    env.Append(MSVC_FLAGS=["/utf-8"])
+
     # ANGLE
     angle_root = os.getenv("ANGLE_SRC_PATH")
     env.Prepend(CPPPATH=[angle_root + "/include"])

--- a/platform/windows/detect.py
+++ b/platform/windows/detect.py
@@ -222,8 +222,8 @@ def configure_msvc(env, manual_msvc_config):
     ## Compile/link flags
 
     env.AppendUnique(CCFLAGS=["/MT", "/Gd", "/GR", "/nologo"])
-    if int(env["MSVC_VERSION"].split(".")[0]) >= 14:  # vs2015 and later
-        env.AppendUnique(CCFLAGS=["/utf-8"])
+    # Force to use Unicode encoding
+    env.Append(MSVC_FLAGS=["/utf-8"])
     env.AppendUnique(CXXFLAGS=["/TP"])  # assume all sources are C++
     if manual_msvc_config:  # should be automatic if SCons found it
         if os.getenv("WindowsSdkDir") is not None:

